core/src/main/java/org/opensearch/sql/lang/PPLLangSpec.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.lang;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+
+/**
+ * PPL language specification implementation.
+ *
+ * <p>This class provides a singleton implementation of {@link LangSpec} for PPL. It defines a
+ * custom mapping from expression types to PPL type names.
+ */
+public class PPLLangSpec implements LangSpec {
+
+  public static final PPLLangSpec PPL_SPEC = new PPLLangSpec();
+
+  private static Map<ExprType, String> exprTypeToPPLType = new HashMap<>();
+
+  static {
+    exprTypeToPPLType.put(ExprCoreType.BYTE, "tinyint");
+    exprTypeToPPLType.put(ExprCoreType.SHORT, "smallint");
+    exprTypeToPPLType.put(ExprCoreType.INTEGER, "int");
+    exprTypeToPPLType.put(ExprCoreType.LONG, "bigint");
+  }
```

Comment:
Since this is a breaking change, why don't we directly change the old type to the new type, instead of introducing `LangSpec`? Isn't this unnecessarily adding complexity?

Reply:
I agree. The core type was not upgraded because I intended for the data type changes to affect only PPL and not SQL. This PR focuses on unifying PPL data types, while SQL data types can be addressed in a separate issue, as changes there would impact JDBC, ODBC, and CLI.

Ideally, the query engine should use well-defined data types, with LangSpec serving as the protocol for translating these engine types to language-specific types. Once the Calcite implementation is complete, CalciteDataType will translate to ExprDataType, and LangSpec will translate from ExprDataType to the PPL response data type.